app/src/main/java/com/github/libretube/ui/models/HomeViewModel.kt

```diff
@@ -0,0 +1,186 @@
+package com.github.libretube.ui.models
+
+import android.content.Context
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.libretube.api.PlaylistsHelper
+import com.github.libretube.api.RetrofitInstance
+import com.github.libretube.api.SubscriptionHelper
+import com.github.libretube.api.obj.Playlists
+import com.github.libretube.api.obj.StreamItem
+import com.github.libretube.constants.PreferenceKeys.HIDE_WATCHED_FROM_FEED
+import com.github.libretube.constants.PreferenceKeys.SAVE_FEED
+import com.github.libretube.db.DatabaseHelper
+import com.github.libretube.db.DatabaseHolder
+import com.github.libretube.db.obj.PlaylistBookmark
+import com.github.libretube.enums.ContentFilter
+import com.github.libretube.helpers.LocaleHelper
+import com.github.libretube.helpers.PlayerHelper
+import com.github.libretube.helpers.PreferenceHelper
+import com.github.libretube.util.deArrow
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+
+class HomeViewModel: ViewModel() {
+
+    private val useSavedFeed get() = PreferenceHelper.getBoolean(SAVE_FEED, false)
+    private val hideWatched get() = PreferenceHelper.getBoolean(HIDE_WATCHED_FROM_FEED, false)
+
+    val trending: LiveData<List<StreamItem>?> get() = _trending
```

Comment:
Why aren't we directly accessing the `MutableLiveData` from below?

Reply:
I created the extra properties to prevent the users of `HomeViewModel` from posting data on the object. This way, we ensure only the view model is responsible for posting new values.

Reply:
We're not a library, so nobody should ever get the idea to post data from anywhere else than the view model (except if it's really necessary), I think we should stick with using `MutableLiveData` only everywhere for consistency across different view models.

Reply:
Yeah, I understand your point, removed the `LiveData` properties.